boromir674/software-patterns
# Software Patterns v2.0.0 Release

2.0.0 (2022-11-19)
==================

Changes
-------

feature
"""""""
- allow flexible adaptation of Proxy pattern with a single entrypoint

test
""""
- fix for Python 3.11

documentation
"""""""""""""
- fix docs
- update architecture diagrams assets that get render on Documentation Website

refactor
""""""""
- fix type checking with mypy on python3.6
- pass type checking in Python3.6
- remove Handler Prototype Pattern
- clean code
- static type checking with mypy of test code (along with prod code)

ci
""
- update image runner to ubuntu-2004 2022 image
- fix type check
- run Build-n-Test Job in Python 3.11
- fix Test Job

breaking
""""""""
- Handler Pattern is no longer supported. 
- Removed the ProxySubject type.
